tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/store/debugger_reducers_test.ts

```diff
@@ -156,6 +238,141 @@ describe('Debugger reducers', () => {
+  ]) {
+    it('Updates alerts data on alertsOfTypeLoaded: empty initial state', () => {
+      const firstAlertExecutionIndex = 10;
+      const state = createDebuggerState({
+        activeRunId: '__default_debugger_run__',
+        alerts: createAlertsState({
+          alertsLoaded: {
+            state: DataLoadState.LOADING,
+            lastLoadedTimeInMs: null,
+          },
+        }),
+        executions: createDebuggerExecutionsState({
+          displayCount,
+          scrollBeginIndex: 0,
+        }),
+      }); // `alerts` state is in an empty initial state.
+      const alert0 = createTestInfNanAlert({
+        op_type: 'RealDiv',
+        execution_index: firstAlertExecutionIndex,
+      });
+      const alert1 = createTestInfNanAlert({
+        op_type: 'Log',
+        execution_index: firstAlertExecutionIndex + 1,
+      });
+      const nextState = reducers(
+        state,
+        actions.alertsOfTypeLoaded({
+          numAlerts: 2,
+          alertsBreakdown: {
+            InfNanAlert: 2,
+          },
+          begin: 0,
+          end: 2,
+          alertType: 'InfNanAlert',
+          alerts: [alert0, alert1],
+        })
+      );
+      expect(nextState.alerts.alertsLoaded.state).toBe(DataLoadState.LOADED);
+      expect(nextState.alerts.alertsLoaded.lastLoadedTimeInMs).toBeGreaterThan(
+        0
+      );
+      expect(nextState.alerts.numAlerts).toBe(2);
+      expect(nextState.alerts.alertsBreakdown).toEqual({
+        [AlertType.INF_NAN_ALERT]: 2,
+      });
+      expect(Object.keys(nextState.alerts.alerts)).toEqual([
+        AlertType.INF_NAN_ALERT,
+      ]);
+      const alertsOfType = nextState.alerts.alerts[AlertType.INF_NAN_ALERT];
+      expect(Object.keys(alertsOfType).length).toBe(2);
+      expect(alertsOfType[0]).toEqual(alert0);
+      expect(alertsOfType[1]).toEqual(alert1);
+      expect(Object.keys(nextState.alerts.executionIndices)).toEqual([
+        AlertType.INF_NAN_ALERT,
+      ]);
+      const executionIndices =
+        nextState.alerts.executionIndices[AlertType.INF_NAN_ALERT];
+      expect(executionIndices).toEqual([
+        firstAlertExecutionIndex,
+        firstAlertExecutionIndex + 1,
+      ]);
+      // Verify that the first alert is scrolled into view.
+      expect(nextState.executions.scrollBeginIndex).toEqual(
+        expectedScrollBegin
+      );
+    });
+  }
+
+  it('Updates alerts data on alertsOfTypeLoaded: non-empty initial state', () => {
+    const alert0 = createTestInfNanAlert({
+      op_type: 'RealDiv',
+      execution_index: 10,
+    });
+    const alert1 = createTestInfNanAlert({
+      op_type: 'Log',
+      execution_index: 11,
+    });
+    const state = createDebuggerState({
+      activeRunId: '__default_debugger_run__',
+      alerts: createAlertsState({
+        alertsLoaded: {
+          state: DataLoadState.LOADING,
+          lastLoadedTimeInMs: null,
+        },
+        numAlerts: 1,
+        alertsBreakdown: {[AlertType.INF_NAN_ALERT]: 1},
```

Comment:
also consider including a test with other types to make sure we are not overwriting the whole alertsBreakdown.